streams/shared/src/main/scala/zio/stream/ZPipeline.scala

```diff
@@ -1094,19 +1094,9 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
    */
   def dropUntilZIO[Env, Err, In](
     p: In => ZIO[Env, Err, Boolean]
-  )(implicit trace: Trace): ZPipeline[Env, Err, In, In] = {
-    lazy val loop: ZChannel[Env, Err, Chunk[In], Any, Err, Chunk[In], Any] = ZChannel.readWithCause(
-      (in: Chunk[In]) =>
-        ZChannel.unwrap(in.dropUntilZIO(p).map { leftover =>
-          val more = leftover.isEmpty
-          if (more) loop else ZChannel.write(leftover) *> ZChannel.identity[Err, Chunk[In], Any]
-        }),
-      (e: Cause[Err]) => ZChannel.refailCause(e),
-      (_: Any) => ZChannel.unit
-    )
-
-    new ZPipeline(loop)
-  }
+  )(implicit trace: Trace): ZPipeline[Env, Err, In, In] =
```

Comment:
A bit outside the scope of this PR but since we're at it:

When I first read `dropUntil` and `dropUntilZIO` I assumed that the method would drop _until but not inclusive_ of the element where the predicate evaluated true. Might be good to change the scaladoc to mention that it will drop elements up to (inclusive) of the element where it evaluated true

Reply:
@kyri-petrou , since this text appears multiple times in the scaladocs, let's agree on the wording here, then I'm willing to make the change. either in this pr or a separate one.